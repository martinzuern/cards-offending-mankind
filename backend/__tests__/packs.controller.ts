import request from 'supertest';
import Server from '../server';

describe('Packs', () => {
  it('should get all packs', (done) =>
    request(Server)
      .get('/api/v1/packs')
      .expect('Content-Type', /json/)
      .expect((r) => {
        expect(r.body).toBeInstanceOf(Array);
        expect(r.body).toHaveLength(72);
        expect(r.body[0]).toEqual({
          abbr: 'BaseUK',
          description: 'The UK edition of the base game',
          icon: '',
          name: 'Base Set (UK Edition)',
          official: true,
          promptsCount: 90,
          responsesCount: 460,
        });
      })
      .end(done));
});
