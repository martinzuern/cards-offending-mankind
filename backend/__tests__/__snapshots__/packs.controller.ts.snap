// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Packs should get all packs 1`] = `
Object {
  "abbr": "Base-INTL",
  "name": "Base Set (INTL Version)",
  "official": true,
  "promptsCount": 89,
  "responsesCount": 510,
}
`;
